matchbox_socket/src/webrtc_socket/socket.rs

```diff
@@ -160,25 +160,26 @@ impl WebRtcSocketBuilder {
     /// The returned [`MessageLoopFuture`] should be awaited in order for messages to be sent and received.
     pub fn build(self) -> (WebRtcSocket, MessageLoopFuture) {
         if self.channels.is_empty() {
-            panic!("You need to configure at least one channel in WebRtcSocketConfig");
+            panic!("You need to configure at least one channel in WebRtcSocketBuilder");
         }
 
-        let (messages_from_peers_tx, messages_from_peers) =
-            new_senders_and_receivers(&self.channels);
         let (peer_state_tx, peer_state_rx) = futures_channel::mpsc::unbounded();
-        let (peer_messages_out_tx, peer_messages_out_rx) =
-            new_senders_and_receivers(&self.channels);
-
+        let (channels, inner_channels): (_, Vec<_>) = (0..self.channels.len())
+            .map(|_| {
+                let (channel, inner_channel) = WebRtcChannel::new();
+                (Some(channel), inner_channel)
+            })
+            .unzip();
```

Comment:
This became more confusing to me

Reply:
Me too, sadly `unzip` only works for `Pair`s :roll_eyes: Might use a macro